### ISSUE TITLE
Set values instead of Random

### ISSUE THREAD

**Issue by zoiwillxx** [challenert/digimon_world_randomizer#35]
Hi, I just found out about this and I don't know if the creators are still active on this project, but I would love to set Digimon values instead of randomizing them. 

Is there a way to do that? Most specifically I would want to set Digimon Stats, Tech Stats, training gains, and max lifespan. 

If you have a forum or a better place to make those requests and questions, please forgive me.

**Comment by challenert** [comment]
I am not really actively developing the tool.  All of those things are probably possible, and for some I already read/write that data, so no ROM research would be required.  When you say Digimon Stats, are you referring to the starting stats of the partner digimon?  If so, that info is read/write accessible already.  I have not done the research for altering max lifespan or training gains.  Both would be code changes (which really just comes down to a ROM offset location, but research in that area is harder than script).  

The biggest part of this would probably be designing/writing some sort of interface to enter all of this information.  Either via the existing config file or some other method, it would require fairly extensive changes to make this possible.

The software is open source, if you are python-familiar or willing to learn.  Adding some of the features you're interested in wouldn't be too bad.

**Comment by zoiwillxx** [comment]
Can you teach me how to make a simple Set script using your tool, like you did with Devimon: 

```
    def updateEvolutionStats( self ):
        """
        Updates evolution requirements for evolving to certain
        digimon.
        """

        devimon = None

        for digi in self.digimonData:
            if( digi.name != 'Devimon' ):
                continue

            devimon = digi
            break

        devimon.evoStats[ 0 ] = 1500
        devimon.evoStats[ 1 ] = 2000
        devimon.evoStats[ 2 ] = 250
        devimon.evoStats[ 3 ] = 100
        devimon.evoStats[ 4 ] = 150
        devimon.evoStats[ 5 ] = 200

        self.logger.logChange( 'Set Devimon stat gains to: 1500  2000  250  100  150  200' )
```

Is it possible to use a similar script to set individual Tech power and accuracy? or other digievolutions stat gain?

**Comment by challenert** [comment]
If you look at `randomizeTechData` you can see me setting all of the tech-related values.  I do it dynamically/randomly in this one, but you could just as easily set specific values.

`
        gigaFreeze = None

        for tech in self.techData:
            if( tech.name != 'Giga Freeze' ):
                continue

            gigaFreeze = tech
            break

        gigaFreeze.power = 999
        gigaFreeze.mp3 = 1
        gigaFreeze.accuracy = 100
        gigaFreeze.effect = 1
        gigaFreeze.effChance = 80
`

For other stat gain, you can simply change the name we're scanning for with `if( digi.name != 'Devimon' ):`

However I should warn you that I think the stat gain read/write is bugged right now.

**Comment by zoiwillxx** [comment]
Whenever I try to build digimon_randomize.py with my code changes, I get a digimon_randomize.exe with only half of the size of one of yours and it does not run at all. 

How can I build it correctly or is there any other way to run the tool using my alterations?

**Comment by challenert** [comment]
You can also run ‘digimon_randomize.py’ directly of you have python
installed. The build probably isn’t working properly due to one or more
python packages not being available on your computer. When you run the
python file directly you may see more helpful errors.

**Comment by zoiwillxx** [comment]
I'm getting KeyError: 'general' when I try to run digimon_randomize.py. 


"digimon_randomize.py", line 24, in <module>
    verbose = config[ 'general' ][ 'LogLevel' ]

raise KeyError(key)
KeyError: 'general'

**Comment by challenert** [comment]
Do you have `settings.ini` in the same directory as
`digimon_randomize.py`?  It looks like it isn't finding your settings file
properly.

**Comment by zoiwillxx** [comment]
Yes it's on the same folder, I even tried to put the path of the file after -settings

**Comment by challenert** [comment]
Ah, you're running into issues caused by me working on the GUI.  If you
checkout back a few commits you could avoid that.  Otherwise yeah, you'd
need to run `digimon_randomize.py -settings settings.ini` and make sure the
`Input` and `Output` are correct as well.

**Comment by zoiwillxx** [comment]
I've manage to fix it, sorry. I tried to open another CMD as administrator and then used cd "path" and run it, instead of using shift+left click + run CMD on folder

**Comment by challenert** [comment]
Okay, glad you figured out what the problem was.
